src/element/mod.rs

```diff
@@ -453,6 +466,13 @@ impl Element {
         &self.value
     }
 
+    /// Returns the source location metadata (row, column).
+    /// If element was constructed from a `LazyValue` then it returns `Some(_)`.
```

Comment:
A couple of doc suggestions:
* I don't think library users necessarily care about whether it was constructed from LazyValue. What's more meaningful to say is that if it was read from Ion _text_, then it will _probably_ have a location.
* Maybe also mention that this is intended to help applications surface errors to users, and so it is not guaranteed to be present in all cases, and should not be relied upon for programmatic manipulation of the source of the Ion data.